ascii.py

```diff
@@ -5,65 +5,62 @@
 from math import floor
 from PIL import Image
 
+TERMINAL_DIMENSIONS = [180, 41, 10, 22]
+NOTEPAD_PLUS_BIG_DIMENSIONS = [400, 100, 10, 18]
+GITHUB_DIMENSIONS = [223, 40, 10, 29]
+DIMENSIONS = GITHUB_DIMENSIONS
+
 def grayscale(img):
-    # Convert img to grayscale.
+    "Convert img to grayscale."
     return img.convert('RGB').convert('L')
 
+
 def average_val(pixels, y, x, y_step, x_step):
-    # Return average value in pixels numpy array in the given rectangle
-    # Expects rectangle coordinates to be in the range of the pixels array
+    """Return average value in boolean pixels numpy array in the given rectangle
+    Expects rectangle coordinates to be in the range of the pixels array."""
     if y_step == 0:
         raise Exception('y_step must be non-zero')
     if x_step == 0:
         raise Exception('x_step must be non-zero')
 
-    start_y = int(floor(y))
-    end_y = int(floor(y+y_step))
-    start_x = int(floor(x))
-    end_x= int(floor(x+x_step))
+    end_y = y + y_step + 1
+    end_x = x + x_step + 1
```

Comment:
this +1 business makes me think you have a fence counting error elsewhere in your code, but i couldn't locate it.

Reply:
rounding error was mostly caused by me trying to be too clever with passing floats into everything. now coordinates are rounded to ints ASAP - no more magical -1's